ql/src/java/org/apache/hadoop/hive/ql/parse/TezCompiler.java

```diff
@@ -1566,13 +1569,38 @@ private void removeSemijoinsParallelToMapJoin(OptimizeTezProcContext procCtx)
 
       List<ExprNodeDesc> keyDesc = selectedMJOp.getConf().getKeys().get(posBigTable);
       ExprNodeColumnDesc keyCol = (ExprNodeColumnDesc) keyDesc.get(0);
-
-      tsProbeDecodeCtx = new TableScanOperator.ProbeDecodeContext(mjCacheKey, mjSmallTablePos,
-          keyCol.getColumn(), selectedMJOpRatio);
+      String realTSColName = getOriginalTSColName(selectedMJOp, keyCol.getColumn());
+      if (realTSColName != null) {
+        tsProbeDecodeCtx = new TableScanOperator.ProbeDecodeContext(mjCacheKey, mjSmallTablePos,
+                realTSColName, selectedMJOpRatio);
+      } else {
+        LOG.warn("ProbeDecode could not find TSColName for ColKey {} with MJ Schema {} ", keyCol, selectedMJOp.getSchema());
```

Comment:
You could throw an error only if you are running in test mode:
```
if (conf.getBoolVar(ConfVars.HIVE_IN_TEST)) {
...
```
While I am not a huge fan of interleaving a check like this within the production code, it may help you identify any issues, gaps, or regressions in the future.
If you do not want to do that, an alternative is uploading a patch throwing an error instead of printing the warning to get a complete test run, then create follow-up issues if there are any. It will not help you with regressions in the future, but you will be able to identify any existing issue.

Reply:
Hey @jcamachor , thanks for the comments!
The HIVE_IN_TEST trick could work as long we enable probedecode optimisation by default right? (currently if off)

Just enabled the optimisation for this PR (throwing an exception instead of warn) to identify any existing issues.

Reply:
Qtest results here:
http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-1152/4/tests/

Seems that for for existing MJ ops the probedecode optimisation works fine (properly finds original TS col alias as well). Not sure if we want to enable probe by default however. Thoughts? cc @ashutoshc

Reply:
Cool! I think we should enable it by default indeed.